declarative-examples/simple-examples/postConditionOrder.groovy

```diff
@@ -0,0 +1,28 @@
+pipeline {
+    // no agent required to run here. All steps run in flyweight executor on Master
+    agent none
+
+    stages {
+        stage("foo") {
+            steps {
+                echo "hello"
+            }
+        }
+    }
+    post {
+    // these steps will run at the end of the pipeline based on the condition.
+    // The conditions run in this order regardless of their arrangement
```

Comment:
Also, an example using all possible `post` conditions would help. The `postInStage` one already has two that aren't here, and it's not clear whether they slot into the order where they appear in the source.